zebrad/src/components/mempool/storage/eviction_list.rs

```diff
@@ -0,0 +1,138 @@
+//! [`EvictionList`] represents the transaction eviction list with
+//! efficient operations.
+use std::{
+    collections::{HashMap, VecDeque},
+    time::{Duration, Instant},
+};
+
+use zebra_chain::transaction;
+
+/// An eviction list that allows to efficiently add entries, get entries,
+/// and remove older entries in the order they were inserted.
+pub struct EvictionList {
+    // Maps each TXID in the list to the most recent instant they were added.
+    unique_entries: HashMap<transaction::Hash, Instant>,
+    // The entries in the order they were inserted.
+    // This can be larger than `unique_entries` if a same txid is added
+    // multiple times. Its instant will be overwritten in
+    // `unique_entries` but all entries will kept in `unique_entries`.
```

Comment:
Incorrect documentation / typo: All entries are kept in `ordered_entries`.

```suggestion
    // `unique_entries` but all entries will kept in `ordered_entries`.
```